security-privacy-questionaire.md

```diff
@@ -0,0 +1,71 @@
+
+### 2.5. Does this specification introduce new state for an origin that persists across browsing sessions?
+
+Yes, this specification lets websites store handles they've gotten access to (via a file or directory picker) in IndexedDB. User agents could also persist the permission grants that go with these handles, but at least in the Chrome implementation these permission grants will only be persistent for installed PWAs. The drive-by web will only have enough state to allow it to re-prompt for access, but the access itself won't be persistent.
+
+Furthermore the user will be able to clear storage (storage is just IndexedDB) and/or revoke permissions to clear the state that was persisted, similarly to how other permissions work.
+
+### 2.6. What information from the underlying platform, e.g. configuration data, is exposed by this specification to an origin?
+
+Anything that exists on disk in files could be exposed by the user to the web. However user agents are encouraged to block list certain directories with particularly sensitive files, and thus somewhat restrict which files and directories the user is allowed to select. For example things like Chrome's "Profile" directory, and other platform configuration data directories are likely going to be on this block list.
+
+### 2.7. Does this specification allow an origin access to sensors on a user’s device
+
+No (unless a device exposes such sensors as "fake" files or directories).
+
+### 2.8. What data does this specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.
+
+The data this specification lets a user expose to an origin is identical to the data exposed via `<input type=file>` and `<input type=file webkitdirectory>`. The differences are in giving the origins the ability to re-prompt for access to files they previously had access to (if handles were stored in IndexedDB), and in the ability for origins to write back to the files (after explicit permission is granted for that).
+
+### 2.9. Does this specification enable new script execution/loading mechanisms?
+
+No.
+
+### 2.10. Does this specification allow an origin to access other devices?
+
+Not really. The exception would be devices that are exposed as "fake" files or directories by the platform. I.e. network shares or cloud storage sync clients could expose data on other devices in a way that looks like regular files or directories. The user agent could let the user pick these files or directories, thereby giving an origin implicit access to this other device. This API doesn't have any functionality to let a website enumerate all network shares on the local network, only explicitly selected files or directories can be accessed by an origin.
```

Comment:
nit: we can remove `fake` here. It's immaterial if it's a byte sequence on disk or affecting a device or the network.

We can also mention how we mitigate this, by re-iterating that we're going to implement a block-list in Chrome.

Reply:
agreed that fake is meaningless here. Not sure how a block list helps here, as I don't think we'd try to block this anyway.

Reply:
The block-list could be helpful in not allowing the API to access `/dev`